r/tests/testthat/test-dplyr-join.R

```diff
@@ -338,3 +346,30 @@ test_that("arrow dplyr query can join two datasets", {
     }
   )
 })
+
+test_that("full joins handle keep", {
+  full_data_df <- tibble::tibble(
+    x = rep(c("a", "b"), each = 5),
+    y = rep(1:5, 2),
+    z = rep("zzz", 10),
+    index = 1:10
+  )
+  small_dataset_df <- tibble::tibble(
+    value = c(0.1, 0.2, 0.3, 0.4, 0.5),
+    x = c(rep("a", 3), rep("b", 2)),
+    y = 1:5,
+    z = 6:10
+  )
+  full_data <- Table$create(full_data_df)
+  small_dataset <- Table$create(small_dataset_df)
+
+  for (keep in c(TRUE, FALSE)) {
+    result <- full_join(small_dataset, full_data, by = c("y", "x"), keep = !!keep) %>%
+      arrange(index) %>%
+      collect()
+    expected <- full_join(small_dataset_df, full_data_df, by = c("y", "x"), keep = !!keep) %>%
+      arrange(index) %>%
+      collect()
+    expect_equal(result, expected)
+  }
```

Comment:
You should be able to use compare_dplyr_binding here too. The data you're joining will get converted to Table automatically

```suggestion

  for (keep in c(TRUE, FALSE)) {
    compare_dplyr_binding(
      .input %>%
        full_join(full_data_df, by = c("y", "x"), keep = !!keep) %>%
        arrange(index) %>%
        collect(),
      small_dataset_df
    )
  }
```